torch/distributed/rpc/internal.py

```diff
@@ -58,6 +67,14 @@ def serialize(self, obj):
         p = pickle.Pickler(f)
         p.dispatch_table = self._dispatch_table
 
+        # rpc api could accept user picklers inheriting from _InternalRPCPickler to serialize rref,
+        # user picklers could have different initialization function from _InternalRPCPickler,
+        # but all the user picklers should call serialize() and use _rref_reducer to pickle rref
+        # in python. also, when _internalRPCPickler is imported to rpc/api.py, rpc.RRef is not 
+        # compiled yet, it is not good place to acces rpc.RRef inside _InternalRPCPickler constructor,
```

Comment:
What determine the order of compilation? Does it work if we explicitly do `from distributed.rpc import RRef` in this file?

Reply:
from .. import .. seems does not work, we can not import rpc in this file as rpc depends on this file, there is circular import issue.